Polygon_mesh_processing/include/CGAL/Side_of_triangle_mesh.h

```diff
@@ -181,6 +181,27 @@ class Side_of_triangle_mesh
     box = tree.bbox();
   }
 
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+  /**
+  * Constructor moving an instance of Side_of_triangle_mesh to a new memory
+  * location with minimal memory copy.
+  * @param other The instance to be moved
+  */
+  Side_of_triangle_mesh(Side_of_triangle_mesh&& other) :
+    tm_ptr{other.tm_ptr},
+    opt_vpm{std::move(other.opt_vpm)},
+    own_tree{other.own_tree},
+    box{other.box},
```

Comment:
I would prefer to see `std::move` called on all members. I agree with the current implementation a simple copy will be made anyway, but we might change the definition of some of those members, one day: for example `tm_ptr` could become a `std::unique_ptr`.

And I think, to be correct, the move constructor and move assignment operator must deal with all members, including `ray_functor` and `vector_functor`, so that the code is correct even if those functors have data members.